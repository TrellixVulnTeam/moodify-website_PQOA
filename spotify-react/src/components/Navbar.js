import React from "react";
import Logout from "./navbar-components/Logout";

export default function Navbar(props) {
    return (<nav className="container flex py-3 p-20 justify-between rounded dark:bg-gray-800" data-testid="navbar">
            <div className={""}>
                <h3 className="text-4xl font-medium text-blue_purple">MOODIFY</h3>
            </div>
            <div className="space-x-3">

                {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
                <a className="text-lg font-semibold whitespace-nowrap text-white"><Logout logout={props.logout}/></a>
            </div>
        </nav>
    )
}