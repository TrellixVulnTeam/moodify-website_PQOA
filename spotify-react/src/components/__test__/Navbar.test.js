import React from 'react';
import ReactDOM from 'react-dom'
import Navbar from "../Navbar";

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer"


it("renders without crashing", ()=> {
    const div = document.createElement("div");
    ReactDOM.render(<Navbar/>, div)
})

afterEach(cleanup);

it("renders navbar correctly", ()=>{
    const {getByTestId} = render(<Navbar />)
    expect(getByTestId("navbar")).toBeDefined()
})

it("matches snapshot", ()=>{
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
})