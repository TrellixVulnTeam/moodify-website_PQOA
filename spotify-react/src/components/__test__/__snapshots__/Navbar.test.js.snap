// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="container flex py-3 p-20 justify-between rounded dark:bg-gray-800"
  data-testid="navbar"
>
  <div
    className=""
  >
    <h3
      className="text-4xl font-medium text-blue_purple"
    >
      MOODIFY
    </h3>
  </div>
  <div
    className="space-x-3"
  >
    <a
      className="text-lg font-semibold whitespace-nowrap text-white"
    >
      <div>
        <button
          className="h-10 px-5 text-red-100 transition-colors duration-150 bg-red-700 rounded-lg focus:shadow-outline hover:bg-red-800"
        >
          Logout
        </button>
      </div>
    </a>
  </div>
</nav>
`;
